File: p13/Entry1.py
class Handler:
    def collide(self, obj1, obj2):
        if isinstance(obj1, Ship) and isinstance(obj2, Station):
            # logic1
            pass
        elif isinstance(obj1, Station) and isinstance(obj2, Asteroid):
            # logic2
            pass
        elif isinstance(obj1, Ship) and isinstance(obj2, Comet):
            # logic3
            pass

File: p13/Entry1.java
interface GameObject{ }
class Ship extends GameObject{}
class Station extends GameObject{}
class Commet extends GameObject{}
class Aestroid extends GameObject{}

class Handler
{
     public void Collide(GameObject go1,GameObject go2)
     {
          if(type(o1)==type(Ship) && type(o2)==type(Station))
          {
            //logic1
          }
           if(type(o1)==type(Station) && type(o2)==type(Aestroid))
          {
            //logic2
          }
           if(type(o1)==type(Ship) && type(o2)==type(Commet))
          {
            //logic3
          }
     }
}
